app/models/sms/answer_hierarchy.rb

```diff
@@ -0,0 +1,64 @@
+class Sms::AnswerHierarchy
+  def initialize
+    # mapping from qing group ID -> answer group
+    @answer_groups = {}
+  end
+
+  def lookup(qing_group)
+    @answer_groups[qing_group.id]
+  end
+
+  def answer_group_for(qing)
```

Comment:
Need to add a spec where the first question in the submitted SMS is from a nested group. This should fail currently since we are not recursively creating the parents all the way up.